libevmasm/AssemblyItem.cpp

```diff
@@ -322,6 +326,27 @@ ostream& solidity::evmasm::operator<<(ostream& _out, AssemblyItem const& _item)
 	return _out;
 }
 
+size_t AssemblyItem::opcodeCount() const noexcept
+{
+	switch (m_type)
+	{
+		case AssemblyItemType::AssignImmutable:
+			// Append empty items if this AssignImmutable was referenced more than once.
+			// For n immutable occurrences the first (n - 1) occurrences will
+			// generate 5 opcodes and the last will generate 3 opcodes,
+			// because it is reusing the 2 top-most elements on the stack.
+			if (!m_immutableOccurrences)
+				// AssignImmutable without any uses does not need to be assigned to at all.
```

Comment:
Please add a mechanism that ensures that `assemble()` is called (and thus `m_immutableOccurences` is properly assigned) before `opcodeCount` or `bytesRequired`, at least for cases that are important.